### ISSUE TITLE
Type check broke "Generate IP Addresses" solvers

### ISSUE THREAD

**Issue by fdellwing** [danielyxie/bitburner#4197]
Version: 2.1.0 (8f4636cb)

Some recent update broke all (?) solvers for "Generate IP Addresses" contracts. The new version introduced this check:

https://github.com/danielyxie/bitburner/blob/fa099c22ab9bbc1412291ae0fb07c80ddf63e0f0/src/data/codingcontracttypes.ts#L511

In the past the game was happy to accept a JSArray as valid solution.

There was no real reason for this breaking change and the original behaviour should be restored.

**Comment by quacksouls** [comment]
Duplicate of #4171.  Fixed by PR #4172.  In the meantime, you can convert your array to a string.  If `arr` is an array that contains your solution, do `arr.toString()` as a temporary fix until #4172 is merged in an upcoming release.

**Comment by quacksouls** [comment]
> I know that PR, but how would it fix the issue? Automatically submitting a JSArray will still fail that check.

Apply the patch at #4172.  Use this script to solve IP contracts:
```js
function all_ip(data) {
    const ret = [];
    for (let a = 1; a <= 3; ++a) {
        for (let b = 1; b <= 3; ++b) {
            for (let c = 1; c <= 3; ++c) {
                for (let d = 1; d <= 3; ++d) {
                    if (a + b + c + d === data.length) {
                        const A = parseInt(data.substring(0, a), 10);
                        const B = parseInt(data.substring(a, a + b), 10);
                        const C = parseInt(data.substring(a + b, a + b + c), 10);
                        const D = parseInt(data.substring(a + b + c, a + b + c + d), 10);
                        if (A <= 255 && B <= 255 && C <= 255 && D <= 255) {
                            const ip = [A.toString(), ".", B.toString(), ".", C.toString(), ".", D.toString()].join("");
                            if (ip.length === data.length + 3) {
                                ret.push(ip);
                            }
                        }
                    }
                }
            }
        }
    }
    return ret;
}

export async function main(ns) {
    // The file name of the coding contract.
    const cct = ns.args[0];
    // The hostname of the server where the coding contract is located.
    const host = ns.args[1];
    // Solve the coding contract.
    const string = ns.codingcontract.getData(cct, host);
    const result = ns.codingcontract.attempt(
        all_ip(string), cct, host, { returnReward: true }
    );
    ns.tprint(result);
}
```
If the patch does not fix the problem as advertised, please comment on ticket #4172.
